homeassistant/components/websocket_api/http.py

```diff
@@ -71,22 +73,37 @@ def __init__(self, hass: HomeAssistant, request: web.Request) -> None:
     async def _writer(self) -> None:
         """Write outgoing messages."""
         # Exceptions if Socket disconnected or cancelled by connection handler
-        with suppress(RuntimeError, ConnectionResetError, *CANCELLATION_ERRORS):
-            while not self.wsock.closed:
-                if (process := await self._to_write.get()) is None:
-                    break
-
-                if not isinstance(process, str):
-                    message: str = process()
-                else:
-                    message = process
-                self._logger.debug("Sending %s", message)
-                await self.wsock.send_str(message)
-
-        # Clean up the peaker checker when we shut down the writer
-        if self._peak_checker_unsub is not None:
-            self._peak_checker_unsub()
-            self._peak_checker_unsub = None
+        to_write = self._to_write
+        logger = self._logger
+        wsock = self.wsock
+        try:
+            with suppress(RuntimeError, ConnectionResetError, *CANCELLATION_ERRORS):
+                while not self.wsock.closed:
+                    if (process := await to_write.get()) is None:
+                        return
+                    message = process if isinstance(process, str) else process()
+
+                    if not self.command_phase or to_write.empty():
```

Comment:
We need to let the client opt-in to this new functionality? Otherwise things like Node-Red or the mobile apps won't work anymore.

Reply:
Added a `supported_features` call to tell hass which features the websocket connection supports